test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/Streaming/ODataJsonLightStreamWritingTests.cs

```diff
@@ -67,6 +67,79 @@ public void CanWritePrimitivePropertyValue()
+        public void CanWritePrimitivePropertyInstanceAnnotationWithoutPropertyValue()
+        {
+            string expectedPayload = String.Format(resourcePayload,
+                ",\"age@Custom.StartAnnotation\":123"
+                );
+
+            RunTest((ODataWriter writer) =>
+            {
+                ODataPrimitiveValue primitiveValue = new ODataPrimitiveValue(123);
+                ODataPropertyInfo propertyInfo = new ODataPropertyInfo { Name = "age" };
+                propertyInfo.InstanceAnnotations.Add(new ODataInstanceAnnotation("Custom.StartAnnotation", primitiveValue));
+
+                // write some properties within the resource and others separately
+                writer.WriteStart(resource);
+
+                writer.WriteStart(propertyInfo);
+                writer.WriteEnd(); // property
+
+                writer.WriteEnd(); // resource
+            },
+            expectedPayload);
+        }
+
+        [Fact]
+        public void CanWritePrimitivePropertyInstanceAnnotationWithPropertyValue()
+        {
+            string expectedPayload = String.Format(resourcePayload,
+                ",\"age@Custom.StartAnnotation\":123,\"age\":37"
+                );
+
+            RunTest((ODataWriter writer) =>
+            {
+                ODataPrimitiveValue primitiveValue = new ODataPrimitiveValue(123);
+                ODataPropertyInfo propertyInfo = new ODataPropertyInfo { Name = "age" };
+                propertyInfo.InstanceAnnotations.Add(new ODataInstanceAnnotation("Custom.StartAnnotation", primitiveValue));
+
+                // write some properties within the resource and others separately
+                writer.WriteStart(resource);
+
+                writer.WriteStart(propertyInfo);
+                    writer.WritePrimitive(new ODataPrimitiveValue(37));
+                writer.WriteEnd(); // property
+
+                writer.WriteEnd(); // resource
+            },
+            expectedPayload);
+        }
+
+        [Fact]
+        public void CanWriteDynamicPropertyWithInstanceAnnotationsButWithoutPropertyValue()
+        {
+            string expectedPayload = String.Format(resourcePayload,
+                 ",\"aDynamicProperty@Custom.StartAnnotation\":123"
+                 );
+
+            RunTest((ODataWriter writer) =>
+            {
+                ODataPrimitiveValue primitiveValue = new ODataPrimitiveValue(123);
+                ODataPropertyInfo propertyInfo = new ODataPropertyInfo { Name = "aDynamicProperty" };
+                propertyInfo.InstanceAnnotations.Add(new ODataInstanceAnnotation("Custom.StartAnnotation", primitiveValue));
+
+                // write some properties within the resource and others separately
+                writer.WriteStart(resource);
+
+                writer.WriteStart(propertyInfo);
+                writer.WriteEnd(); // property
+
+                writer.WriteEnd(); // resource
+            },
+            expectedPayload);
+        }
+

```

Comment:
@xuzhg There at least 3 places where the new `ReadNestedPropertyInfoWithoutValue` is called. Do the two tests cover all 3 calls?

Reply:
@xuzhg The async tests for `ODataJsonLightReader` are here https://github.com/OData/odata.net/blob/master/test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightReaderTests.cs

Reply:
@gathogojr I added async test cases to verify the async scenarios. Please take a look again.